src/common/checksum.c

```diff
@@ -87,11 +88,35 @@ daos_cont_csum_prop_is_enabled(uint16_t val)
 {
 	if (val != DAOS_PROP_CO_CSUM_CRC16 &&
 	    val != DAOS_PROP_CO_CSUM_CRC32 &&
-	    val != DAOS_PROP_CO_CSUM_CRC64)
+	    val != DAOS_PROP_CO_CSUM_CRC64 &&
+	    val != DAOS_PROP_CO_CSUM_SHA1 &&
+	    val != DAOS_PROP_CO_CSUM_SHA256 &&
+	    val != DAOS_PROP_CO_CSUM_SHA512)
 		return false;
 	return true;
 }
 
+int
+daos_str2csumcontprop(const char *value)
+{
+	if (!strcmp(value, "off"))
+		return DAOS_PROP_CO_CSUM_OFF;
+	else if (!strcmp(value, "crc16"))
```

Comment:
I assume this is a one time comparison?  Is value properly vetted?   Do we need to use strncmp?

This big if/else if/else if ... makes me want to think of a clever way.   One really hacky option is to cast each to an int * and to a switch on the integer value of the first 4 bytes of each :-)...but that would be unreadable.   Or check only the first character and 4th characters.  But I digress...what you have is readable.

Reply:
Good catch, I will use strncmp instead. I actually tried a couple different approaches first and then just copied this from what was already in daos.c. I might take another look :-).